src/Frontend/Page/Admin.hs

```diff
@@ -461,6 +469,36 @@ adminSettingsGaPClassesEdit :: ActionPersist m => SchoolClass -> m PageAdminSett
 adminSettingsGaPClassesEdit clss =
     PageAdminSettingsGaPClassesEdit clss <$> query (getUsersInClass clss)
 
+instance FormPage PageAdminSettingsGaPUserDelete where
+    type FormPagePayload PageAdminSettingsGaPUserDelete = ()
+
+    type FormPageResult PageAdminSettingsGaPUserDelete = ()
+
+    formAction (PageAdminSettingsGaPUserDelete user) = U.Admin $ U.AdminDeleteUser (user ^. _Id)
+
+    redirectOf _ _ = U.Admin $ U.AdminAccess PermUserView
+
+    makeForm _ = pure ()
+
+    formPage _v form p@(PageAdminSettingsGaPUserDelete user) =
+        adminFrame p . semanticDiv p . form $ do
+            p_ "Are you sure???"
+            a_ [href_ . U.Admin $ U.AdminEditUser (user ^. _Id), class_ "btn-cta"] "Cancel"
+            DF.inputSubmit "Nutzer löschen"
+
+    guardPage _ = do
+        role <- view userRole <$> currentUser
+        -- FIXME: Log invalid user deletion attempt.
+        pure $ if role /= Admin
+                then Just . relPath $ U.Logout
+                else Nothing
```

Comment:
I am strongly in favour of removing this guard.

This is an ad-hoc security check.  Security needs to be done more rigorously, and once we get around to do that, this will be redundant..

So far our threat model is *nobody ever does anything destructive, not even by accident*.